stellargraph/layer/rgcn.py

```diff
@@ -0,0 +1,598 @@
+                constraint=self.bias_constraint,
+            )
+        else:
+            self.bias = None
+
+        self.built = True
+
+    def call(self, inputs):
+        """
+        Applies the layer.
+
+        Args:
+            inputs (list): a list of 2 + R input tensors that includes
+                node features (size 1 x N x F),
+                output indices (size 1 x M),
+                and a graph adjacency matrix (size N x N) for each relationship.
+                R is the number of relationships in the graph (edge type),
+                N is the number of nodes in the graph, and
+                F is the dimensionality of node features.
+
+        Returns:
+            Keras Tensor that represents the output of the layer.
+        """
+        features, out_indices, *As = inputs
+        batch_dim, n_nodes, _ = K.int_shape(features)
+        if batch_dim != 1:
+            raise ValueError(
+                "Currently full-batch methods only support a batch dimension of one"
+            )
+
+        # Remove singleton batch dimension
+        features = K.squeeze(features, 0)
+        out_indices = K.squeeze(out_indices, 0)
+
+        # Calculate the layer operation of RGCN
+        output = K.dot(features, self.self_kernel)
+
+        for i in range(self.num_relationships):
+            h_graph = K.dot(As[i], features)
+            output += K.dot(h_graph, self.relational_kernels[i])
+
+        # Add optional bias & apply activation
+        if self.bias is not None:
+            output += self.bias
+        output = self.activation(output)
+
+        # On the final layer we gather the nodes referenced by the indices
+        if self.final_layer:
+            output = K.gather(output, out_indices)
+
+        # Add batch dimension back if we removed it
+        if batch_dim == 1:
+            output = K.expand_dims(output, 0)
+
+        return output
+
+
+class RGCN:
+    """
+    A stack of Relational Graph Convolutional layers that implement a relational graph
+    convolution neural network model as in https://arxiv.org/pdf/1703.06103.pdf
+
+    The model minimally requires specification of the layer sizes as a list of ints
+    corresponding to the feature dimensions for each hidden layer,
+    activation functions for each hidden layers, and a generator object.
+
+    To use this class as a Keras model, the features and pre-processed adjacency matrix
+    should be supplied using the :class:`RelationalFullBatchNodeGenerator` class.
+    The generator object should be instantiated as follows::
+
+        generator = RelationalFullBatchNodeGenerator(G)
+
+    Note that currently the RGCN class is compatible with both sparse and dense adjacency
+    matrices and the :class:`RelationalFullBatchNodeGenerator` will default to sparse.
+
+    For more details, please see the RGCN demo notebook:
+
+    emos/node-classification/rgcn/rgcn-aifb-node-classification-example.ipynb
```

Comment:
I think you mean "demos/...", missing the letter d.

Reply:
nice catch, I'll quickly fix that